### ISSUE TITLE
ELI5: How do I install this on windows?

### ISSUE THREAD

**Issue by Gfdshui** [openai/jukebox#242]
A pre-compiled exe file or similar would be greatly appreciated.

I know it's possible to install this on windows, but the explanations aren't detailed enough and outdated, and installing this normally is error after error. ELI5 please. 

GPU: NVIDIA GeForce GTX 1080 Ti
OS: Windows 10
Regular Python version w/o conda: 3.8.0
Python version when conda is installed: 3.9.5
Using the default 'conda create --name jukebox python=3.7.5'
wget.exe in PATH
Can't use Google Collab as I don't have Collab Pro. Google doesn't give out GPU's with specs that support jukebox.

This isn't the focus of this post, but the problems I have got are: (combined issues from openai/jukebox and peterlazzarino/jukebox-windows, using Git Bash and CMD, not a complete list of the errors I got)
- RuntimeError: Failed to initialize NCCL
- .cache models file 'vqvae.pth.tar' didn't install correctly and it's size was 0 bytes. After pre-installing vqvae.pth.tar and attempting to make a sample, 'prior_level_0.pth.tar' file is made with 0 bytes
- 'TypeError: ord() expected a character, but string of length 0 found' after trying to follow 
-  https://github.com/openai/jukebox/issues/36#issuecomment-624084129

**Comment by nbinobied** [comment]
- `RuntimeError: Failed to initialize NCCL` please check my comment https://github.com/openai/jukebox/issues/216#issuecomment-917455337
- For this, I believe you would have to copy the link in the cmd that its trying to download from (i think it was openai.azure or something similar) and download it manually then move it to the cache folder (you can see the path in cmd). Heads up, they are large files. https://github.com/openai/jukebox/issues/36#issuecomment-625796530
- Not sure about this, maybe its related to the file downloaded.
- I was able to sample from their models successfully after 12hours of debugging.

I just started training the vqvae after another 6 hours of debugging, heard it take a week or 2. I'm training on 1GB of music around 2 hours of Qanun (Arabic Instrument) for an experiment for my course work. I'll update when it finishes.

**Comment by cicinwad** [comment]
> A pre-compiled exe file or similar would be greatly appreciated.
> 
> I know it's possible to install this on windows, but the explanations aren't detailed enough and outdated, and installing this normally is error after error. ELI5 please.
> 
> GPU: NVIDIA GeForce GTX 1080 Ti OS: Windows 10 Regular Python version w/o conda: 3.8.0 Python version when conda is installed: 3.9.5 Using the default 'conda create --name jukebox python=3.7.5' wget.exe in PATH Can't use Google Collab as I don't have Collab Pro. Google doesn't give out GPU's with specs that support jukebox.
> 
> This isn't the focus of this post, but the problems I have got are: (combined issues from openai/jukebox and peterlazzarino/jukebox-windows, using Git Bash and CMD, not a complete list of the errors I got)
> 
> * RuntimeError: Failed to initialize NCCL
> * .cache models file 'vqvae.pth.tar' didn't install correctly and it's size was 0 bytes. After pre-installing vqvae.pth.tar and attempting to make a sample, 'prior_level_0.pth.tar' file is made with 0 bytes
> * 'TypeError: ord() expected a character, but string of length 0 found' after trying to follow
> * [Does this work on windows? #36 (comment)](https://github.com/openai/jukebox/issues/36#issuecomment-624084129)

Dude, use windows 8, it works on windows 8 only
